### PR TITLE
[WIP] Issue #2543390 by bojanz, joshtaylor: Provide a cart view mode for ea…

### PR DESCRIPTION
**Pull request by joshuataylor**
…ch relevant entity type

### CODE REVIEW

File: modules/cart/commerce_cart.module [hunk 1]
```diff
@@ -4,3 +4,30 @@
  * @file
  * Implements the shopping cart system and add to cart features.
  */
+use Drupal\Core\Entity\Entity\EntityViewMode;
+
+/**
+ * Implements hook_rebuild().
+ */
+function commerce_cart_rebuild() {
+
```
Comment (bojanz) [inline modules/cart/commerce_cart.module]: Unneeded new line.

File: modules/cart/commerce_cart.module [hunk 2]
```diff
@@ -4,3 +4,30 @@
  * @file
  * Implements the shopping cart system and add to cart features.
  */
+use Drupal\Core\Entity\Entity\EntityViewMode;
+
+/**
+ * Implements hook_rebuild().
```
Comment (bojanz) [inline modules/cart/commerce_cart.module:11]: Below the Implements line let's describe what this function does. "Creates a cart view mode for each line item source entity type."

File: modules/cart/commerce_cart.module [hunk 3]
```diff
@@ -4,3 +4,30 @@
...
+function commerce_cart_rebuild() {
+
+  $entityTypes = \Drupal::service('entity.manager')->getDefinitions();
+  $entityTypes = array_filter($entityTypes, function ($entityType) {
+    return $entityType->isSubclassOf('\Drupal\commerce\LineItemSourceInterface');
+  });
+
+  foreach ($entityTypes as $entityType) {
+    /** @var \Drupal\commerce\LineItemSourceInterface $entityType */
+    $cartViewModeExists = EntityViewMode::load($entityType->id() . '.cart');
```
Comment (bojanz) [inline modules/cart/commerce_cart.module]: $cartViewModeExists can be just $exists for brevity, it's clear enough.

**Comment by joshuataylor** [comment]
Updated PR, added dbolinovski's code, gave him credit.

**Comment by bojanz** [comment]
Committed in f2d874806929956acfda1a5fad664adac1226116.

**Pull request closed without merging**